app/ui/src/app/customizations/api-connector/api-connector-detail.component.ts

```diff
@@ -1,30 +1,33 @@
 import { Component, OnInit } from '@angular/core';
-import { ActivatedRoute, Router } from '@angular/router';
-import { Observable } from 'rxjs/Observable';
+import { ActivatedRoute } from '@angular/router';
 
-import { ApiConnector, ApiConnectorData } from './api-connector.models';
+import { ApiConnectorData } from './api-connector.models';
 import { ApiConnectorStore } from './api-connector.store';
+import { ApiConnectorService } from '@syndesis/ui/customizations/api-connector/api-connector.service';
 
 @Component({
   selector: 'syndesis-api-connector-detail',
   templateUrl: 'api-connector-detail.component.html',
   styleUrls: ['api-connector-detail.component.scss']
 })
 export class ApiConnectorDetailComponent implements OnInit {
-  apiConnectorData$: Observable<ApiConnectorData>;
-  loading$: Observable<boolean>;
+  apiConnector: ApiConnectorData;
+  loading: boolean;
 
   constructor(private apiConnectorStore: ApiConnectorStore,
-              private router: Router,
+              private apiConnectorService: ApiConnectorService,
               private route: ActivatedRoute) {
-    this.loading$ = this.apiConnectorStore.loading;
-    this.apiConnectorData$ = this.apiConnectorStore.resource;
+    this.loading = true;
   }
 
   ngOnInit() {
-    this.route.paramMap
-      .first(params => params.has('id'))
-      .map(params => params.get('id'))
-      .subscribe(id => this.apiConnectorStore.load(id));
+    const id = this.route.snapshot.paramMap.get('id');
+
+    if (id) {
+      this.apiConnectorService.getApiConnector(id).subscribe(apiConnectorData => {
+        this.apiConnector = apiConnectorData;
+        this.loading = false;
+      });
+    }
   }
```

Comment:
You can even take a more formal _Reactive_ approach for this, by removing the conditional and relying on RxJS operators:

```typescript
  ngOnInit() {
    this.route.paramMap
      .first(params => params.has('id'))
      .map(params => params.get('id'))
      .switchMap(id => this.apiConnectorService.getApiConnector(id))
      .subscribe(apiConnectorData => {
        this.apiConnector = apiConnectorData;
        this.loading = false;
      });
  }
```